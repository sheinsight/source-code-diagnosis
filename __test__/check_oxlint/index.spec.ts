import { expect, test } from "vitest";
import { checkOxlint } from "../../index";
import { fileURLToPath } from "node:url";
import path from "node:path";
import { dirname } from "node:path";

const __filename = fileURLToPath(import.meta.url);

test("No phantom dependencies", async () => {
	const cwd = path.resolve(dirname(__filename), "fixtures");
	const res = await checkOxlint(
		JSON.stringify({
			rules: {
				"unicorn/no-instanceof-array": "error",
			},
		}),
		{ cwd },
	);

	console.log(res.at(0)?.labels);
});
