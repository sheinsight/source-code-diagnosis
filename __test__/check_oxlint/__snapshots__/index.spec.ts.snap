// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`No phantom dependencies 1`] = `
[
  {
    "code": "eslint-plugin-unicorn(no-instanceof-array)",
    "fileName": "index.js",
    "help": "The instanceof Array check doesn't work across realms/contexts, for example, frames/windows in browsers or the vm module in Node.js.",
    "labels": [
      {
        "loc": {
          "end": {
            "col": 27,
            "line": 2,
          },
          "start": {
            "col": 5,
            "line": 2,
          },
        },
        "span": {
          "length": 22,
          "offset": 14,
        },
      },
    ],
    "message": "Use \`Array.isArray()\` instead of \`instanceof Array\`.",
    "severity": "error",
    "url": "https://oxc.rs/docs/guide/usage/linter/rules/unicorn/no-instanceof-array.html",
  },
]
`;
